compiler/cli/src/org/jetbrains/kotlin/cli/common/klibArguments.kt

```diff
@@ -99,6 +96,25 @@ private fun parseCustomKotlinAbiVersion(customKlibAbiVersion: String?, collector
     return KotlinAbiVersion(version[0], version[1], version[2])
 }
 
+private fun getZipFileSystemAccessor(arguments: CommonKlibBasedCompilerArguments): ZipFileSystemAccessor {
+    val cacheLimit = arguments.klibZipFileAccessorCacheLimit.toIntOrNull()
+    if (cacheLimit != null && cacheLimit > 0) {
+        return DisposableZipFileSystemAccessor(cacheLimit)
+    } else {
+        return ZipFileSystemInPlaceAccessor
+    }
+}
+
+private class DisposableZipFileSystemAccessor private constructor(
```

Comment:
There is no need to additionally mark the constructor as private. The whole class is already private.